pmlc/conversion/stdx_to_llvm/stdx_to_llvm.cc

```diff
@@ -345,6 +345,42 @@ struct CmpXchgLowering : public LoadStoreOpLowering<stdx::AtomicRMWOp> {
   }
 };
 
+struct FPToSILowering : public LLVMLegalizationPattern<stdx::FPToSIOp> {
+  using LLVMLegalizationPattern<stdx::FPToSIOp>::LLVMLegalizationPattern;
+  using Base = LLVMLegalizationPattern<stdx::FPToSIOp>;
+
+  PatternMatchResult
+  matchAndRewrite(Operation *op, ArrayRef<Value> operands,
+                  ConversionPatternRewriter &rewriter) const override {
+    if (auto fptosiOp = cast<stdx::FPToSIOp>(op)) {
```

Comment:
`cast` can never fail, it only has an `assert` internally. But I don't think we need to actually downcast in this case.